quantum/pointing_device/pointing_device_scroll.c

```diff
@@ -0,0 +1,357 @@
+/* Copyright 2022 Alabastard
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifdef POINTING_DEVICE_SCROLL_ENABLE
+
+#    include "pointing_device_scroll.h"
+
+/* initialize static functions */
+
+/* duplication of inline function from pointing_device.c */
+static inline int8_t pointing_device_hv_clamp(mouse_xy_report_t value) {
+    if (value < INT8_MIN) {
+        return INT8_MIN;
+    } else if (value > INT8_MAX) {
+        return INT8_MAX;
+    } else {
+        return value;
+    }
+}
+
+static report_mouse_t process_scroll_mode(scroll_status_t scroll_status, report_mouse_t mouse_report);
+
+// set up context and functions if using two pointing devices
+#    if defined(SPLIT_POINTING_ENABLE) && defined(POINTING_DEVICE_COMBINED)
+/* set up local context for storing current values */
+static scroll_context_t scroll_context = {config.is_left = SCROLL_LEFT_SIDE_DEFAULT, .config.tg_mode = SCROLL_MODE_DEFAULT, .status.mode = SCROLL_MODE_DEFAULT};
```

Comment:
Shouldn't this be .config.is_left (with the dot)?

Reply:
yes again this was a typo on my part sorry about this as I hadn't tested that this code could compile yet but thank you for doing so.